src/main/java/org/openlmis/buq/web/buq/BottomUpQuantificationController.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org.
+ */
+
+package org.openlmis.buq.web.buq;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.openlmis.buq.domain.buq.BottomUpQuantification;
+import org.openlmis.buq.domain.buq.BottomUpQuantificationLineItem;
+import org.openlmis.buq.domain.buq.BottomUpQuantificationStatusChange;
+import org.openlmis.buq.dto.buq.BottomUpQuantificationDto;
+import org.openlmis.buq.dto.buq.BottomUpQuantificationLineItemDto;
+import org.openlmis.buq.dto.buq.BottomUpQuantificationStatusChangeDto;
+import org.openlmis.buq.exception.NotFoundException;
+import org.openlmis.buq.i18n.MessageKeys;
+import org.openlmis.buq.repository.buq.BottomUpQuantificationRepository;
+import org.openlmis.buq.service.buq.BottomUpQuantificationService;
+import org.openlmis.buq.util.Pagination;
+import org.openlmis.buq.web.BaseController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * Controller used to expose Bottom-Up Quantifications via HTTP.
+ */
+@Controller
+@RequestMapping(BottomUpQuantificationController.RESOURCE_PATH)
+@Transactional
```

Comment:
In my opinion marking controllers `@Transactional` is an anti-pattern

Reply:
I stuck to the OpenLMIS convention (@Transactional in controllers, none in services)

Reply:
Marked as resolve per slack discussion.